### PR TITLE
Geode-7498: show index type

### PR DESCRIPTION
**Pull request by jmelchio**
co-authored-by: Dale Emery <demery@pivotal.io>
co-authored-by: Jinmei Liao <jliao@pivotal.io>

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [ ] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

**Comment by jmelchio** [comment]
@dschneider-pivotal @demery-pivotal @jinmeiliao

File: geode-management/src/main/java/org/apache/geode/management/configuration/IndexType.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  public String getName() {
+    return this.name;
+  }
+
+  public static IndexType valueOfSynonym(String name) {
+    name = name.toUpperCase();
+
+    switch (name) {
+      case "KEY":
+        return valueOf("PRIMARY_KEY");
```
Comment (demery-pivotal) [inline geode-management/src/main/java/org/apache/geode/management/configuration/IndexType.java]: Consider `return PRIMARY_KEY;` Similar for the other values.

File: geode-management/src/main/java/org/apache/geode/management/configuration/IndexType.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  HASH_DEPRECATED("HASH"),
+  PRIMARY_KEY("KEY");
+
+  private String name;
+
+  IndexType(String name) {
+    this.name = name;
+  }
+
+  public String getName() {
```
Comment (demery-pivotal) [inline geode-management/src/main/java/org/apache/geode/management/configuration/IndexType.java]: `name` is an unfortunate name for this property, given that every enum already has a `name()` method. It's and confusing awkward that `name()` and `getName()` return different values.

Below, we call this `synonym`. Consider changing the field to `synonym` and this method to `getSynonym()`.

Comment (jinmeiliao) [reply geode-management/src/main/java/org/apache/geode/management/configuration/IndexType.java]: if we don't define the synonyms inside IndexType class, then I agree with you. If the IndexType is aware of the synonyms, then it's the responsibility of the class itself to provide a translation.

Comment (jinmeiliao) [reply geode-management/src/main/java/org/apache/geode/management/configuration/IndexType.java]: agree!

**Comment by jinmeiliao** [comment]
if we don't define the synonyms inside IndexType class, then I agree with you. If the IndexType is aware of the synonyms, then it's the responsibility of the class itself to provide a translation.

> The mapping between the new `IndexType` enum and the old names is needed only by the converter. I think it's a mistake to put that mapping into `IndexType`, and to make the mapping visible via a method (`valueOfSynonym()`). Putting that method into `IndexType` makes the mapping part of the public API.

File: geode-management/src/main/java/org/apache/geode/management/configuration/IndexType.java (hunk 1 above)
Comment (jmelchio) [reply geode-management/src/main/java/org/apache/geode/management/configuration/IndexType.java]: Thanks @demery-pivotal I applied this change.

File: geode-management/src/main/java/org/apache/geode/management/configuration/IndexType.java (hunk 2 above)
Comment (jmelchio) [reply geode-management/src/main/java/org/apache/geode/management/configuration/IndexType.java]: @demery-pivotal I've applied this change.

**Pull request merged**